packages/e2e/puppeteer/tests/location-component.ts

```diff
@@ -1,54 +1,159 @@
 import puppeteer from "puppeteer";
 import { flows, host, puppeteerOptions } from "../config";
 import { FlowOptions } from "../types";
-import { random } from 'vergunningcheck-mocking/src/util';
-import { getFixturesByProperties, NO_MONUMENT, NO_CITY_SCAPE } from 'vergunningcheck-mocking/src/restriction';
+import { random } from "vergunningcheck-mocking/src/util";
+import {
+  getFixturesByProperties,
+  NO_MONUMENT,
+  NATIONAL_MONUMENT,
+  MUNICIPAL_MONUMENT,
+  NO_CITY_SCAPE,
+  MUNICIPAL_CITY_SCAPE,
+  NATIONAL_CITY_SCAPE,
+  uniqueFixtures,
+} from "vergunningcheck-mocking/src/restriction";
 
+// Disable the warning "MaxListenersExceededWarning: Possible EventEmitter memory leak detected"
+// See: https://github.com/puppeteer/puppeteer/issues/594
+process.setMaxListeners(Infinity);
+
+// Test the flow with a random address to see if all checkers are working
 const addresses = getFixturesByProperties([NO_MONUMENT, NO_CITY_SCAPE]);
-export const address = random(addresses);
-if (!address) {
-  throw new Error('address not found');
+export const randomAddress = random(addresses);
+if (!randomAddress) {
+  throw new Error("address not found");
 }
+const testFlow = async (page: puppeteer.Page) => {
+  await page.waitForSelector('[data-testid="next-button"]');
+  await page.click('[data-testid="next-button"]');
+
+  await page.waitForSelector('[name="postalCode"]');
+  await page.type('[name="postalCode"]', randomAddress[0]);
+  await page.type('[name="houseNumberFull"]', randomAddress[1].toString());
+
+  await page.waitForSelector('[data-testid="location-found"]', {
+    visible: true,
+  });
+};
+
+flows.forEach(async (flow) => {
+  flow.checkers.forEach(async (url) => {
+    const browser = await puppeteer.launch(puppeteerOptions);
+    try {
+      const page = await browser.newPage();
+      await page.goto(`${host}/${url}`);
+      await testFlow(page);
+    } catch (e) {
+      console.error(e);
+    } finally {
+      await browser.close();
+    }
+  });
+});
 
 // @TODO actually click in dropdown
 // await page.type('[name="houseNumberFull"]', '19');
 // await page.click('[data-testid="suggestList"] a');
 
-const testFlow = async (page: puppeteer.Page, options: FlowOptions) => {
-
+// Test with multipe addresses the visiblity of the restrictions on the page
+const testRestrictions = async (
+  page: puppeteer.Page,
+  options: FlowOptions,
+  address: any
+) => {
   await page.waitForSelector('[data-testid="next-button"]');
   await page.click('[data-testid="next-button"]');
 
   await page.waitForSelector('[name="postalCode"]');
   await page.type('[name="postalCode"]', address[0]);
   await page.type('[name="houseNumberFull"]', address[1].toString());
 
-  await page.waitForSelector('[data-testid="location-found"]', { visible: true });
+  await page.waitForSelector('[data-testid="location-found"]', {
+    visible: true,
+  });
+
+  const momumentType = address[2][0];
+  const cityScapeType = address[2][1];
```

Comment:
This is very error-prone. We need to have an alternative for this. Let's discuss offline.

Reply:
Fixed